docs/user_guide/expand_path.rst

```diff
@@ -24,47 +21,64 @@ data spec.
+        },
+        "phy" : {
+            "base_directory": "/path/to/processed_data"
+            "folder_path": "{subject_id}/{session_id}/phy"
         }
     }
 
+    # Instantiate LocalPathExpander
+    path_expander = LocalPathExpander()
+
     # Expand paths and extract metadata
     metadata_list = path_expander.expand_paths(source_data_spec)
 
     # Print the results
     for metadata in metadata_list:
         print(metadata)
 
-In the example above, we first create an instance of :py:class:`~neuroconv.tools.path_expansion.LocalPathExpander`.
-We then specify a source data spec in the form of a dictionary. The dictionary specifies the interface, base
-directory, and f-string format for the file path. We then call the ``expand_paths`` method with the source data spec,
-which returns a list of :py:class:`~neuroconv.utils.dict.DeepDict` objects that contain the metadata extracted from
-the file paths that match the f-string format.
+The ``expand_paths`` method returns a list of :py:class:`~neuroconv.utils.dict.DeepDict` objects that contain two
+dictionaries: ``source_data`` and ``metadata``. The ``source_data`` dictionary contains the resolved filepath of each
+interface, while the ``metadata`` dictionary contains the metadata extracted from the filepaths. Currently, only
+``subject_id``, ``session_id``, and ``session_start_time`` are supported, but this approach could in principle be
+extended to support extraction of more metadata.
 
-The :py:class:`~neuroconv.utils.dict.DeepDict` objects contain two dictionaries: ``source_data`` and ``metadata``. The
-``source_data`` dictionary contains the file path and the interface, while the ``metadata`` dictionary contains the
-metadata extracted from the file path. Currently, only ``subject_id`` and ``session_id`` are supported, but this
-approach could in principle be extended to support extraction of more metadata.
+Specifying Metadata Format
+--------------------------
+The f-string format allows you to constrain the search for more precise metadata matching using the
+`Format Specification Mini-Language`_ and the `1989 C standard format codes`_ for datetimes. Below are some common
+examples.
+
+Length
+~~~~~~
+For example, you might have a data path where the ``subject_id`` and ``session_id`` are next to each other, and they
+can be disambiguated because the ``subject_id`` is always 4 characters and the ``session_id`` is always 5 characters.
+This can be expressed as ``"{subject_id:4}{session_id:5}"``.
+
+Character type
+~~~~~~~~~~~~~~
+If the ``subject_id`` is always numeric, you could use ``n``, e.g. ``"{subject_id:n}"``, which will match only digits.
 
+Datetimes
+~~~~~~~~~
+If your session start time is present in your data path, you can indicate this following the
+`1989 C standard format codes`_ for datetimes. For example, ``"{session_start_time:%Y-%m-%d}"`` will match
+``"2021-01-02"`` and evaluate it to ``datetime.datetime(2021, 1, 2)``.
+
```

Comment:
```suggestion

Real-world Examples
-----------------------

Below are some full examples of how this feature can be used on some organizational patterns taken from real datasets.

Example 1
~~~~~~~

TODO
```

Reply:
let's leave this for later once we have some real-world examples to provide

Reply:
This is actually the main thing I was requesting from this PR; the other subsections on special capabilities are nice, but what we need for the GUIDE docs is some concrete examples of how to use this in practice

Here are some real-world examples

Example 1) [Allen Institute Visual Coding](https://registry.opendata.aws/allen-brain-observatory/)

Nice and simple as a first example

Example file: `/allen-brain-observatory/visual-coding-2p/ophys_movies/ophys_experiment_645691416.h5`

f-string for all such files: `/allen-brain-observatory/visual-coding-2p/ophys_movies/ophys_experiment_{session_id}.h5`


Example 2) [YutaS of Buszaki](https://app.globus.org/file-manager?origin_id=188a6110-96db-11eb-b7a9-f57b2d55370d&origin_path=%2FSenzaiY%2FYMV01%2FYMV01_170818%2F)

Example file: `/SenzaiY/YMV01/YMV01_170818/YMV01_170818.dat`

f-string for all such files: `/SenzaiY/{subject_id}/{subject_id}_{session_start_time:%y%m%d}/{subject_id}_{session_start_time:%y%m%d}.dat`


Example 3) Was thinking to use Froemke's from figshare, but the URL (even the DOI) in this paper no longer lead anywhere? https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7677212/

Do you know how to find that dataset again?

Reply:
ok, this is fine. Want to add it to this PR in rst?

Reply:
> Want to add it to this PR in rst?

If the suggestion is accepted, then I'm fine leaving it to you to decide how best to make the section look really nice

Reply:
@CodyCBakerPhD I've got a lot on my plate. Could you please do it or delegate it to someone else? Otherwise, let's push this PR through as is and format these examples later.

Reply:
come to think of it, @felixp8 , want to take a stab at this?